api/scpca_portal/utils.py

```diff
@@ -92,3 +94,66 @@ def get_csv_zipped_values(
     and returns the zipped values as a list.
     """
     return list(zip(*(data.get(key).split(delimiter) for key in args), strict=True))
+
+
+"""
+The `aws s3 ls <bucket>` command called in `list_s3_paths()` returns a list of two types of entries:
+- Bucket Object Entries
+- Bucket Prefix Entries
+In order to create a standard API, where `entry.file_path` could be accessed
+irrespective of the entry type, we've created two named tuples which follow the return format
+of each of the bucket entry types.
+"""
+BucketObjectEntry = namedtuple("BucketObjectEntry", ["date", "time", "size_in_bytes", "file_path"])
+BucketPrefixEntry = namedtuple("BucketPrefixEntry", ["prefix_designation", "file_path"])
+
+
+def list_s3_paths(
+    relative_path: Path = Path(),
+    *,
+    bucket_path: Path = Path(common.INPUT_BUCKET_NAME),
+    recursive: bool = True,
+):
+    """
+    Queries a path on an inputted s3 bucket
+    and returns bucket's existing content as a list of Path objects.
+
+    The `aws s3 ls <bucket>` command returns a list of two types of entries:
+    - Bucket Object Entries
+    - Bucket Prefix Entries
+    In order to create a standard API, where `entry.file_path` could be accessed
+    irrespective of the entry type, we've created two named tuples which follow the return format
+    of each of the bucket entry types.
+    """
+    root_path = Path().joinpath(*bucket_path.parts, *relative_path.parts)
+    command_inputs = ["aws", "s3", "ls", f"s3://{root_path}"]
+
+    if recursive:
+        command_inputs.append("--recursive")
+
+    if "public" in str(bucket_path):
+        command_inputs.append("--no-sign-request")
+
+    try:
+        result = subprocess.run(command_inputs, capture_output=True, text=True, check=True)
+        output = result.stdout
+    except subprocess.CalledProcessError as error:
+        logger.error(
+            """
+            `{}`: Cause of error not returned, note: folder must exist and be non-empty
```

Comment:
Can you add the command_inputs to the error message?

Reply:
I actually found that the `subprocess.CallProcessError` prints the args (which are the `command_inputs`) when the `print()` function (in this case the logger function) is called on it, so it ends up being redundant.

Reply:
This is what it prints
```
 `Command '['aws', 's3', 'ls', 's3://scpca-portal-public-test-inputs/2024-04-19/SCPCP999992/SCPCP999992_bulk', '--recursive', '--no-sign-request']' returned non-zero exit status 1.`: Cause of error not returned, note: folder must exist and be non-empty 
```